File: algorithms/code/leetcode/lc191_number_of_1_bits/lc191_number_of_1_bits.py
# https://leetcode.com/problems/number-of-1-bits/


class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        for _ in range(0, 32):
            count += n % 2  # Faster than "& 1" in Python
            n >>= 1
        return count


class TestCode:
    def test_11(self) -> None:
        assert Solution().hammingWeight(11) == 3

File: algorithms/code/leetcode/lc191_number_of_1_bits/lc191_number_of_1_bits.go
// Package lc191 implements https://leetcode.com/problems/number-of-1-bits/
package lc191

func hammingWeight(n int) int {
	count := 0
	for range 32 {
		count += n % 2
		n >>= 1
	}
	return count
}
